src/EffectList.ts

```diff
@@ -91,14 +91,18 @@ export class EffectList {
 				typeof effect.config.tickInterval !== 'undefined'
 			) {
 				const now = Date.now();
+				const sinceLastTick = now - effect.state.lastTick;
 				if (
-					Date.now() <
-					effect.state.ticks * effect.config.tickInterval * 1000
+					sinceLastTick <
+					effect.config.tickInterval * 1000
 				) {
 					continue;
 				}
 				effect.state.lastTick = now;
-				effect.state.ticks && effect.state.ticks++;
+				if (!effect.state.ticks) {
```

Comment:
Might be redundant from:
[state.tick=0](https://github.com/Ranvier-TS/core-ts/blob/8aacd73de3e746582d95116f90f9d51e598e8129/src/Effect.ts#L176)

Reply:
I don't want to set it to zero unless it is null-ish (undefined, null, NaN), the redundancy is if it's already set to zero which is probably fine. Thinking it through, I could update the check to be `if (!effect.state.ticks && effect.state.ticks !== 0)`

Reply:
I was just pointing out that most likely the state.ticks would be 0 (unless was set differently) from the Effect constructor. Didn't know if you had seen that.